core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -1745,10 +1745,11 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
    * has overridden the call site using `setCallSite()`, this will return the user's version.
    */
   private[spark] def getCallSite(): CallSite = {
-    val callSite = Utils.getCallSite()
     CallSite(
-      Option(getLocalProperty(CallSite.SHORT_FORM)).getOrElse(callSite.shortForm),
-      Option(getLocalProperty(CallSite.LONG_FORM)).getOrElse(callSite.longForm)
+      Option(getLocalProperty(CallSite.SHORT_FORM))
```

Comment:
I see your point though now this calls `Utils.getCallSite` twice when neither property is set. That might be OK, but I wonder if you can instead retrieve both property values, and then proceed to call  `Utils.getCallSite` once if either is null.